doc/tree-menu.md

```diff
@@ -0,0 +1,45 @@
+# ASAB WebUI TreeMenu component
+
+This component is based on the library [react-simple-tree-menu](https://github.com/iannbing/react-simple-tree-menu).
+
+Displays menu items in a clear structure.
+
+## Example code
+
+```javascript
+import { TreeMenu } from 'asab-webui';
+...
+
+<TreeMenu
+	resource={resource}
+	data={treeData}
+	hasSearch={true}
+	hasNodes={true} // default: false
+	searchOptions={{
+		placeholder: t("Search"),
+		dropdown: {
+			title: t("Actions"),
+			children: ChildrenCustomDropdownComponent
+		}
+	}}
+/>
+
+```
+
+
+## Props:
+
+#### Required:
+
+	- `data`: object or array, data in a special structure. The detailed data structure is described in the library documentation react-simple-tree-menu.
```

Comment:
@aringocode Are you sure that it can be either object or array? What I know it should be an array of objects

Reply:
The documentation says it's either an array or an object:
![image](https://github.com/TeskaLabs/asab-webui/assets/65866093/78b72d61-eeab-46d5-8f8f-18b5a90ee5a3)

Reply:
Oki